report_xlsx/controllers/main.py

```diff
@@ -0,0 +1,106 @@
+                context.update(data["context"])
+            xlsx = report.with_context(**context)._render_xlsx(
+                reportname, docids, data=data
+            )[0]
+            xlsxhttpheaders = [
+                (
+                    "Content-Type",
+                    "application/vnd.openxmlformats-"
+                    "officedocument.spreadsheetml.sheet",
+                ),
+                ("Content-Length", len(xlsx)),
+            ]
+            return request.make_response(xlsx, headers=xlsxhttpheaders)
+        return super(ReportController, self).report_routes(
+            reportname, docids, converter, **data
+        )
+
+    @route()
+    def report_download(self, data, context=None):
+        requestcontent = json.loads(data)
+        url, report_type = requestcontent[0], requestcontent[1]
+        try:
+            if report_type == "xlsx":
+                reportname = url.split("/report/xlsx/")[1].split("?")[0]
+                docids = None
+                if "/" in reportname:
+                    reportname, docids = reportname.split("/")
+                if docids:
+                    # Generic report:
+                    response = self.report_routes(
+                        reportname, docids=docids, converter="xlsx", context=context
+                    )
+                else:
+                    # Particular report:
+                    data = dict(
+                        url_decode(url.split("?")[1]).items()
+                    )  # decoding the args represented in JSON
+                    if "context" in data:
+                        context, data_context = json.loads(context or "{}"), json.loads(
+                            data.pop("context")
+                        )
+                        context = json.dumps({**context, **data_context})
+                    response = self.report_routes(
+                        reportname, converter="xlsx", context=context, **data
+                    )
+
+                report = request.env["ir.actions.report"]._get_report_from_name(
+                    reportname
+                )
+                filename = "%s.%s" % (report.name, "xlsx")
+
+                if docids:
+                    ids = [int(x) for x in docids.split(",")]
+                    obj = request.env[report.model].browse(ids)
+                    if report.print_report_name and not len(obj) > 1:
+                        report_name = safe_eval(
+                            report.print_report_name, {"object": obj, "time": time}
+                        )
+                        filename = "%s.%s" % (report_name, "xlsx")
+                if not response.headers.get("Content-Disposition"):
+                    response.headers.add(
+                        "Content-Disposition", content_disposition(filename)
+                    )
+                return response
+            else:
+                return super(ReportController, self).report_download(data, context)
```

Comment:
Apply PEP 3135 (28-Apr-2007) to "super()."

Reply:
Done